polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -710,7 +710,8 @@ function IndexTableBase({
   const tableClassNames = classNames(
     styles.Table,
     hasMoreLeftColumns && styles['Table-scrolling'],
-    selectMode && styles.disableTextSelection,
+    selectMode &&
+      classNames(styles.disableTextSelection, styles['Table-select-mode']),
```

Comment:
These could be merged together but I like the readability / distinction between the two